torch/nested/_internal/ops.py

```diff
@@ -520,17 +520,33 @@ def _nested_jagged_to_strided(func, *args, **kwargs):
     )
 
 
-register_jagged_func(
+register_jagged_func(torch.ops.aten.detach.default, "self: jt_all")(
+    jagged_unary_pointwise
+)
+
+
+@register_jagged_func(
     [
         torch.ops.aten.empty_like.default,
         torch.ops.aten.ones_like.default,
         torch.ops.aten.zeros_like.default,
-        torch.ops.aten.empty_like.default,
         torch.ops.aten.randn_like.default,
-        torch.ops.aten.detach.default,
     ],
     "self: jt_all",
-)(jagged_unary_pointwise)
+)
+def like_factory_default(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp = new_kwargs.pop("input")
+
+    # Default layout is technically torch.strided but only jagged is supported here.
+    # Rather than force users to specify the layout, assume jagged.
+    # This should be set to strided for redispatching on values.
+    new_kwargs["layout"] = torch.strided
```

Comment:
Is this layout the same as the "layout" in NST?   Does it mean that the layout of NT._values is in strided layout, but the NJT itself is in jagged layout?

Reply:
> Is this layout the same as the "layout" in NST?

yes, NST uses `torch.strided` as its `layout`. The `torch.strided` layout is also the default for standard dense tensors.

> Does it mean that the layout of NT._values is in strided layout, but the NJT itself is in jagged layout?

yep, that's right. both the `values` and `offsets` of an NJT have the strided layout (as they are standard dense tensors), but the NJT itself has `layout=torch.jagged`.